### DISCUSSION TITLE
General Questions about openHASP

### DISCUSSION THREAD

**Discussion by fvanroie** [HASwitchPlate/openHASP#58] [Q&A]
Here you can ask questions related to the use of openHASP. If you are unsure if a particular issue is a bug, please ask here first before opening a new bug-report.

**Comment by sleepypig2003** [comment]
Hi!
Great project, I love it and am planning to use a lanbon8 and m5stack as remote in my homeassistant setup.

I was wondering if there was a way to issue commands based on the object, ie. pressing a button -> issues a command?

ie. if I press a button, then I can display a temperature on a page, if I press another button, I can display a different image, etc...

Thanks!

**Comment by apline87** [comment]
Hi, Thank you very much for this project! I have a question regarding custom build. I have tested my setup and wiring with TFT_eSPI example "Keypad_240x320" it works very well and also other examples. My board is Lolin ESP32 S2 mini and display is ILI9341 with XT2046 touch screen (not lolin, red one). I compiled .bin myself with PlatformIO with success. I made config in platformio_override.ini for board and same pins as in TFT_eSPI example "Keypad_240x320".

When I turn on device i see only OPENHASP text on dark blue background. And Wifi hotspot. I can also reach out device with web interface. and upload pages.json file, when taking screenshoot I can see page layout properly, but on the physical screen still the same OPENHASP text.

My setup:
`[platformio]
extra_configs =

    user_setups/esp32s2/*.ini
   
[override]
; -- Hasp config options --------------------------------------
build_flags =
; -- Uncomment the next line to use the file include/user_config_override.h settings
   -DUSE_CONFIG_OVERRIDE

;region -- Default Build Environments : Used when Build All ---
extra_default_envs =
    s2-mini-esp32s2_ili9341
;endregion

;region -- Define your local COM ports for each environment ---
[env:s2-mini-esp32s2_ili9341]
monitor_port = /dev/cu.usbmodem01       ; Change to the correct port
upload_port = ${env:s2-mini-esp32s2_ili9341.monitor_port}

board_build.partitions = user_setups/esp32/partitions_4MB.csv ; default.csv
debug_tool = esp-prog
debug_init_break = tbreak setup
;endregion
extends = esp32s2_4mb_v2
board = esp32s2_solo1

build_flags =
    ${env.build_flags}
    ${esp32s2.build_flags}

;region -- TFT_eSPI build options ------------------------
    -D ILI9341_DRIVER=1
    -D TFT_WIDTH=240
    -D TFT_HEIGHT=320
    -D TFT_ROTATION=0 ; Use default, see TFT_ROTATION values
    -D SPI_FREQUENCY=40000000
    -D SPI_READ_FREQUENCY=20000000
    ;-D USER_SETUP_LOADED=1
    -D LGFX_USE_V1=1
    -D SUPPORT_TRANSACTIONS
    -D SPI_TOUCH_FREQUENCY=2500000
    -D TOUCH_DRIVER=0x2046     ; XPT2046 Resistive SPI touch panel driver
    -D TFT_MISO=3   ;D6  Use default HSPI
    -D TFT_MOSI=5  ;D7  Use default HSPI
    -D TFT_SCLK=7   ;D5  Use default HSPI
    -D TFT_DC=9    ;D8
    -D TFT_CS=12    ;D0
    -D TFT_BCKL=36  ;None, configurable via web UI (e.g. 2 for D4)
    -D TOUCH_CS=11  ;D3  (can also be D1 or D2)
    -D TOUCH_SDA=-1
    -D TOUCH_SCL=-1
    -D TOUCH_IRQ=-1
    -D I2C_TOUCH_PORT=0
    -D I2C_TOUCH_ADDRESS=0
    -D I2C_TOUCH_FREQUENCY=0
    -D TFT_RST=-1   ;RST
;endregion
;region -- Library options -------------------------------
lib_deps =
    ${env.lib_deps}
    ${esp32s2.lib_deps}
    ${lovyangfx.lib_deps}
lib_ignore =
    ${env.lib_ignore}
    ${esp32s2.lib_ignore}
    ;TFT_eSPI
;endregion`

![20230418_231612](https://user-images.githubusercontent.com/98914420/232895333-0a38916e-a924-4271-b188-4990be88711f.jpg)
I have no idea where to dig. Please help me!
Thank you in advance!

**Reply by fvanroie** [reply]
The ESP32-S2 is not recommended for openHASP. It only has 320kB of PSram while the regular ESP32 and ESP32-S3 have 520kB.
Most likely LVGL fails to initialize because a lack of available memory to create its buffers...

I would use another recommended ESP32 or ESP32-S3 board which has more memory. If you must use an ESP32-S2 then you need to dial in the memory buffers so that they can fit in the available Sram.

**Reply by apline87** [reply]
Thank you very much for very fast reply. Is there any way to get project "lighter" myself?  I don't need all the features. For example compiling with out LVGL library. Or compile previous build (0.6.2) I wondering that esp8266 works almost flawlessly.

Thank you!

**Reply by fvanroie** [reply]
You are free to modify the code in any way you like. However, compiling _without_ LVGL doesn't make sense to me because that's the whole premise around which openHASP was built.

Support for the ESP8266 was already dropped last year and ESP32-S2 hasn't really been supported... You're free to run older builds if they work on your device, but they are not supported anymore going forward. In my opinion it's just not worth the time and effort required to make openHASP run on inferior hardware. Especially since many supported development boards only costs a few dollars.

Maybe the configuration can be tweaked and optimized to make it work on the ESP32-S2 mini... but that's really up to you.
If you want something that *just works* please consider using a regular ESP32 or an ESP32-S3.

**Reply by apline87** [reply]
It makes sense, thank you very much for your answers! Just for understanding, what is the reason to keep board info on "Getting started page" (OpenHasp wiki). Am I correct assuming that with Lolin TFT 2.4 this board should work otherwise I don't understand why to mention it there...
![Screenshot 2023-04-19 at 22 05 49](https://user-images.githubusercontent.com/98914420/233176229-a6f05c9b-d1d2-4ed7-bf7f-fd1831b73100.png)

**Reply by fvanroie** [reply]
As far as I know the S2 is pin-compatible with that board, but feel free to update the wiki page accordingly.

**Comment by apline87** [comment]
Hi, now I've gotten work board with Lolin TFT 2.4 Screen works perfectly, but touch does not work. Also all communication with Home Assistance works and web ui. Any ideas how I can get work the touch screen?

**Comment by ConBeeUser** [comment]
Hi, is it possible to use the OpenHASP platform for a digital photo frame? 

**I would like to have it display photos (jpg) from a web server (or possibly a microSD card).**

And when I swipe to the side or touch the display I would like to show up some controls to control my home assistant instance.

Best regards 
Mat

**Comment by Hedcyber** [comment]
Hello there! Guys i need help. This is the first time i trying to do something with openHasp and cant find the asnwer for my question. And the question is, Where i should placing config code. In these page [https://www.openhasp.com/0.6.3/integrations/home-assistant/sampl_conf/#generic-thermostatclimate](url) we have **Generic thermostat/climate** with UI code and Functionality code or settings, so where i should place this settings code, i can`t figure it out?
